src/Compilers/Core/CodeAnalysisTest/StrongNameProviderTests.cs

```diff
@@ -32,61 +29,62 @@ public void ResolveStrongNameKeyFile()
 
             // search paths
-            var providerSP = new VirtualizedStrongNameProvider(
-                existingFullPaths: fs,
-                searchPaths: ImmutableArray.Create(@"C:\goo", dir, subdir));
+            var searchPathsSP = ImmutableArray.Create(@"C:\goo", dir, subdir);
 
-            path = providerSP.ResolveStrongNameKeyFile(fileName);
+            path = provider.FileSystem.ResolveStrongNameKeyFile(fileName, searchPathsSP);
             Assert.Equal(filePath, path, StringComparer.OrdinalIgnoreCase);
 
             // null base dir, no search paths
-            var providerNullBase = new VirtualizedStrongNameProvider(
-                existingFullPaths: fs,
-                searchPaths: ImmutableArray.Create<string>());
+            var searchPathsEmpty = ImmutableArray<string>.Empty;
 
             // relative path
-            path = providerNullBase.ResolveStrongNameKeyFile(fileName);
+            path = provider.FileSystem.ResolveStrongNameKeyFile(fileName, searchPathsEmpty);
             Assert.Null(path);
 
             // full path
-            path = providerNullBase.ResolveStrongNameKeyFile(filePath);
+            path = provider.FileSystem.ResolveStrongNameKeyFile(filePath, searchPathsEmpty);
             Assert.Equal(filePath, path, StringComparer.OrdinalIgnoreCase);
 
             // null base dir
-            var providerNullBaseSP = new VirtualizedStrongNameProvider(
-                existingFullPaths: fs,
-                searchPaths: ImmutableArray.Create(dir, subdir));
+            var searchPathsNullBaseSP = ImmutableArray.Create(dir, subdir);
 
             // relative path
-            path = providerNullBaseSP.ResolveStrongNameKeyFile(fileName);
+            path = provider.FileSystem.ResolveStrongNameKeyFile(fileName, searchPathsNullBaseSP);
             Assert.Equal(filePath, path, StringComparer.OrdinalIgnoreCase);
 
             // full path
-            path = providerNullBaseSP.ResolveStrongNameKeyFile(filePath);
+            path = provider.FileSystem.ResolveStrongNameKeyFile(filePath, searchPathsNullBaseSP);
             Assert.Equal(filePath, path, StringComparer.OrdinalIgnoreCase);
         }
 
         public class VirtualizedStrongNameProvider : DesktopStrongNameProvider
         {
-            private readonly HashSet<string> _existingFullPaths;
+            private class VirtualStrongNameFileSystem : StrongNameFileSystem
+            {
+                private readonly HashSet<string> _existingFullPaths;
+                public VirtualStrongNameFileSystem(HashSet<string> existingFullPaths)
+                {
+                    _existingFullPaths = existingFullPaths;
+                }
+
+                internal override bool FileExists(string fullPath)
+                {
+                    return fullPath != null && _existingFullPaths != null && _existingFullPaths.Contains(FileUtilities.NormalizeAbsolutePath(fullPath));
```

Comment:
why do we allow _existingFullPaths to be null

Reply:
This was a port of the VirtualizedStrongNameProvider method of the same name.  If you'd like, I could investigate usages and see if I can remove this edge case.